x-pack/legacy/plugins/lens/server/routes/index_stats.ts

```diff
@@ -164,3 +275,170 @@ export function recursiveFlatten(
   });
   return returnTypes;
 }
+
+export async function getNumberHistogram(
+  aggSearchWithBody: (body: unknown) => Promise<unknown>,
+  field: { name: string; type: string; esTypes?: string[] }
+) {
+  const minMaxResult = (await aggSearchWithBody({
+    sample: {
+      sampler: { shard_size: 5000 },
+      aggs: {
+        min_value: {
+          min: { field: field.name },
+        },
+        max_value: {
+          max: { field: field.name },
+        },
+        top_values: {
+          terms: { field: field.name, size: 10 },
+        },
+      },
+    },
+  })) as AggregationSearchResponse<
+    unknown,
+    {
+      body: {
+        aggs: {
+          sample: {
+            sampler: {};
+            aggs: {
+              min_value: {
+                min: { field: string };
+              };
+              max_value: {
+                max: { field: string };
+              };
+            };
+          };
+        };
+      };
+    }
+  >;
+
+  const minValue = minMaxResult.aggregations!.sample.min_value.value;
+  const maxValue = minMaxResult.aggregations!.sample.max_value.value;
+  const terms = minMaxResult.aggregations!.sample.top_values;
+
+  let histogramInterval = (maxValue! - minValue!) / 10;
+
+  if (Number.isInteger(minValue!) && Number.isInteger(maxValue!)) {
+    histogramInterval = Math.ceil(histogramInterval);
+  }
+
+  if (histogramInterval === 0) {
+    return {
+      top_values: terms,
+      histogram: { buckets: [], doc_count: minMaxResult.aggregations!.sample.doc_count },
+    };
+  }
+
+  const histogramResult = (await aggSearchWithBody({
+    sample: {
+      sampler: { shard_size: 5000 },
```

Comment:
Maybe SHARD_SIZE should be a const, since it's currently a magic number that exists in several places in this file.